### DISCUSSION TITLE
Automatic backlinks?

### DISCUSSION THREAD

**Discussion by tscahill** [zim-desktop-wiki/zim-desktop-wiki#1756] [Ideas]
Spending a lot of time pruning and splicing my Zim tree. This post is seeking suggestions an positing one.

Usually, when I cut something from one page to paste into another, I create dual backlinks to allow quickly switching between the new and the old location for the material I moved. Here is the use case currently.

I'm trying to get the religion of the task list. -- I'm still spending half my time working on things that are not on the task list, but that might be another issue :). -- Anyway, I'm wanting to keep track of when I originally created the task (the better to get all forensic on it after), but I'm finding, with an average of 130 items on my task list, that I prefer grouping like tasks on the same page. This to better keep notes and funny stories where they belong. So, for that last few days, 

1. - I'm clicking on items in my task list. 
2. - That almost always takes me to a journal page, or subpage. 
3. - From there, copying the task details, 
4. - then creating an anchor, 
5. - travel to the intended new location, and 
6. - pasting the task there. 
7. - and creating another anchor to the new page
8. - copy that anchor location
9. - Alt-left arrow to return to the original page
10. - Highlight the relevant part of the original task and 
11. - Ctrl-L to paste the link  to the new page's anchor. 
12. - Delete the task list lables and/or check boxes
13. - copy the orignal location anchor, then 
14. - either click the return link or Alt-left arrow to return to the new page, then
15. - Highlight the relevant part of the task
16. - then Ctrl-L to past the link to the old location.

Imagine doing that 130 times. Anybody got a work around? Could automatic backlinks be an exciting new feature?

**Comment by supertoothy** [comment]
Zim had this feature a long time before it became cool. Just go to preferences / Plugins and activate the backlinks pane plugin.

**Reply by tscahill** [reply]
Yes. I have been using the backlinks pane/plugin since before I was cool. :+1: The issue with the current implementation is when you have an extensive page, with multiple links to different parts of it. The backlinks panel shows what links to the page, but it does not show which page matches which link.

As I noted in my explanation, I'm transfering 130 seperate items from their original page(_s_) [Edit] to one consolidated page. It does little good to have a backlinks panel with up to 130 different pages listed. As a work-around, I have tried to make use of Zim's search capability, by placing unique (I hope) markings by each consolidated item. Not ideal.

**Reply by introt** [reply]
> The backlinks panel shows what links to the page, but it does not show which page matches which link.

Related: #1559 (I feel your pain)

**Comment by jaap-karssenberg** [comment]
Maybe you could be more concrete on what you want to automate?  Is it
cut-paste while leaving links?

For moving tasks around, you might also want to use the "migrated" checkbox
type. Have s look at the bullet journal manual page for ideas.

Op vr 10 dec. 2021 20:08 schreef tscahill ***@***.***>:

**Reply by tscahill** [reply]
Cut and paste while leaving links. Yes please. A wee bit of data stickiness. 

Well, what I'm describing might be similar to a audit trail. (Yes! I am also trying to wrap my brain around Zim's revision tracking, although not from the desire to undo something, but more to preserve context) In a linux file system, What I'm doing could be be though of like a file hard link. (In which case, a single version of a hard-linked file exists in the file system, but it appears in more than one directory.) Like a hard link, you only need to change the name of the file once, and it changes everywhere, but I'm not wanting to keep duplicates of the same content dispersed around the wiki. You also might think of what I'm attempting as something like a source footnote. -- Where did that content come from?

So, for example, I have a TODO key word 'READ'. This is great. I now have a very extensive reading list built in zim. Whether I actually read them all is another matter. But already, the question has arisen: why did I want to read _that_? I probably noted a reason on the journal page for the day I added the material to the reading list. But which day was it? What was I thinking?....

And yes, I've been examining the variations possible with the 'migrated' checkbox.  I haven't thought enough yet about how to position that capability vis a vis the ##anchor. [>] and [<] do suggest movement, and seem quite suggestive as structures to hang links upon. However, as you know, the double [[]] involved would be neither a checkmark or a link.

Laughing now as I just found a bug trying to take advantage of Zim's capacity to turn --> into an arrow, and hang a link off that. I suppose I'm veering into reckless and irresponsible terrain now. I'll submit that bug (or is a feature?) then come back to this.
